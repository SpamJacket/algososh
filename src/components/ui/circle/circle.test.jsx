import renderer from "react-test-renderer";
import { Circle } from "./circle";

describe("Рендеринг круга", () => {
  it("Круг без букв рендерится без ошибок", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг c буквами рендерится без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с head рендерится без ошибок", () => {
    const tree = renderer.create(<Circle head={"head"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с react-элементом в head рендерится без ошибок", () => {
    const tree = renderer
      .create(<Circle head={<Circle letter="Тест" isSmall={true} />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с tail рендерится без ошибок", () => {
    const tree = renderer.create(<Circle tail={"tail"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с react-элементом в tai рендерится без ошибок", () => {
    const tree = renderer
      .create(<Circle tail={<Circle letter="Тест" isSmall={true} />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с index рендерится без ошибок", () => {
    const tree = renderer.create(<Circle index={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с пропcом isSmall ===  true рендерится без ошибок", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг в состоянии default рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="default" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг в состоянии changing рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="changing" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг в состоянии modified рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="modified" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
