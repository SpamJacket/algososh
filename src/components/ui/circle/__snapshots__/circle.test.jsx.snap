// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендеринг круга Круг c буквами рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг без букв рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг в состоянии changing рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг в состоянии default рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг в состоянии modified рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг с head рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    head
  </div>
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг с index рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined undefined undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        >
          Тест
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Рендеринг круга Круг с react-элементом в tai рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined undefined undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        >
          Тест
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
</div>
`;

exports[`Рендеринг круга Круг с tail рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    tail
  </div>
</div>
`;

exports[`Рендеринг круга Круг с пропcом isSmall ===  true рендерится без ошибок 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined undefined undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;
